lib/ecto/adapters/firebird/connection.ex

```diff
@@ -826,9 +825,11 @@ defmodule Ecto.Adapters.Firebird.Connection do
       message: "DISTINCT with multiple columns is not supported by Firebird"
   end
 
-  def select(%{select: %{fields: fields}, distinct: distinct} = query, sources) do
+  def select(%{select: %{fields: fields}, distinct: distinct} = query, limit, offset, sources) do
     [
       "SELECT ",
+      limit,
+      offset,
```

Comment:
You should use this way, to combine all lists. And theris no need to pass the limit and the offset as a parameter.

```suggestion
  def select(%{select: %{fields: fields}, distinct: distinct} = query, sources) do
    ["SELECT ", limit(query, sources), offset(query, sources),
```